### PR TITLE
Create ICoreDebugProcess11-interface

### PR DESCRIPTION
**Pull request by mikelle-rogers**
## Summary

Adding files to document ICoreDebugProcess11 and ICorDebugModule4

Fixes [#1374](https://github.com/dotnet/diagnostics/issues/1374)

### CODE REVIEW

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+f1_keywords: 
+  - "ICorDebugModule4"
+helpviewer_keywords: 
+  - "ICorDebugModule4 interface [.NET Framework debugging]"
+ms.assetid:
+  - "apiref"
+---
+# ICorDebugModule4 Interface
+
+Query to see if the module is loaded into memory in mapped/hydrated format.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md:20]:
```suggestion
Provides a method that determines whether the module is loaded into memory in mapped/hydrated format.
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  
+## Methods  
+  
+|Method|Description|  
+|------------|-----------------|  
+|[ICorDebugModule4::IsMappedLayout Method](icordebugmodule4-ismappedlayout-method.md)|Queries to see if the module is loaded into memory in mapped/hydrated format.|  
+  
+## Remarks  
+
+ This interface logically extends the "ICorDebugModule", "ICorDebugModule2", and "ICoreDebugModule3" interfaces.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md:41]:
```suggestion
 This interface logically extends the `ICorDebugModule`, `ICorDebugModule2`, and `ICoreDebugModule3` interfaces.  
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+helpviewer_keywords: 
+  - "IsMappedLayout method, ICorDebugModule4 interface [.NET Framework debugging]"
+  - "ICorDebugModule4::IsMappedLayout method [.NET Framework debugging]"
+ms.assetid:
+topic_type: 
+  - "apiref"
+---
+# ICorDebugModule4::IsMappedLayout Method
+
+Query to see if the module is loaded into memory in mapped/hydrated format.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:22]:
```suggestion
Determines whether a module is loaded into memory in mapped/hydrated format.
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  
+```cpp  
+HRESULT IsMappedLayout(
+      [out] BOOL *pIsMapped
+      );  
+```  
+  
+## Parameters  
+  
+ pIsMapped
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:34]:
```suggestion
`pIsMapped`\
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+```cpp  
+HRESULT IsMappedLayout(
+      [out] BOOL *pIsMapped
+      );  
+```  
+  
+## Parameters  
+  
+ pIsMapped  
+ [out] Pointer to a BOOL to store mapping information. TRUE will represent mapped format while FALSE represents flat format.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:35]:
```suggestion
`[out]` Pointer to a BOOL to store mapping information. TRUE represents mapped format while FALSE represents flat format. 
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+```  
+  
+## Parameters  
+  
+ pIsMapped  
+ [out] Pointer to a BOOL to store mapping information. TRUE will represent mapped format while FALSE represents flat format. 
+  
+## Return Value  
+
+ S_OK
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:39]:
```suggestion
`S_OK`\
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  
+ pIsMapped  
+ [out] Pointer to a BOOL to store mapping information. TRUE will represent mapped format while FALSE represents flat format. 
+  
+## Return Value  
+
+ S_OK  
+ Successfully created the reader.  
+  
+ S_FALSE
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:42]:
```suggestion
`S_FALSE`\
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+ The layout could not be determined.
+  
+## Remarks  
+
+The pIsMapped value set in pIsMapped should only be interpreted as valid when this function returns S_OK. All other return values (including
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:47]:
```suggestion
The `pIsMapped` value should only be interpreted as valid when this function returns `S_OK`. All other return values (including
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+S_FALSE) indicate that the layout could not be determined and pIsMapped should be ignored.
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:48]:
```suggestion
`S_FALSE`) indicate that the layout could not be determined and `pIsMapped` should be ignored.
```

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  
+ **Library:** CorGuids.lib  
+  
+ **.NET Framework Versions:** 4.5, 4, 3.5 SP1  
+  
+## See also
+
+- [ICorDebugRemoteTarget Interface](icordebugremotetarget-interface.md)
+- [ICorDebug Interface](icordebug-interface.md)
+
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-ismappedlayout-method.md:64]:
```suggestion
```

File: docs/framework/unmanaged-api/debugging/icordebugprocess11-enumerateloaderheapmemoryregions-method.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+  - "mscordbi.dll"
+api_type: 
+  - "COM"
+ms.assetid:
+topic_type: 
+  - "apiref"
+---
+# ICorDebugProces11::EnumerateLoaderHeapMemoryRegions Method
+
+[Supported in the .NET Framework 4.5.2 and later versions]
```
Comment (gewarren) [inline docs/framework/unmanaged-api/debugging/icordebugprocess11-enumerateloaderheapmemoryregions-method.md:19]:
```suggestion
[Supported in .NET Framework 4.5.2 and later versions.]  
```

**Review by gewarren — COMMENTED** [review]
They should also be added to the TOC and index as needed.

**Comment by mikelle-rogers** [comment]
image

@gewarren, why does this show up in my pull request? I am not sure what the link is referring to. Do I message those who wrote troubleshoot-app-launch.md?

**Comment by gewarren** [comment]
> image
> 
> @gewarren, why does this show up in my pull request? I am not sure what the link is referring to. Do I message those who wrote troubleshoot-app-launch.md?

This has been fixed in main. If you pull the latest upstream main into your branch, it will resolve the error.

**Comment by gewarren** [comment]
Can you also add them to this index file? https://github.com/dotnet/docs/blob/9a186e3127b632a3dec570c7f67105f44047de96/docs/framework/unmanaged-api/debugging/debugging-interfaces.md

**Review by tommcdon — APPROVED** [review]
Lgtm, thanks!

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+  
+## Requirements  
+
+ **Platforms:** See [System Requirements](../../get-started/system-requirements.md).  
+  
+ **Header:** CorDebug.idl, CorDebug.h  
+  
+ **Library:** CorGuids.lib  
+  
+ **.NET Framework Versions:** 4.5, 4, 3.5 SP1
```
Comment (hoyosjs) [inline docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md:54]: We don't support this in .NET Framework nor do we support `ICorDebugProcess11`. It's gonna fail with E_NOINTERFACE.

Comment (mikelle-rogers) [reply docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md:54]: Will you please explain more what you mean? 
Based off your comment, I will remove all of the docs on ICoreDebugProcess11, is that correct? 
Is there anything else I should do in addition to that?

**Review by noahfalk — COMMENTED** [review]
Oh shoot, I forgot to submit the comment.

File: docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md (hunk 12 above)
Comment (hoyosjs) [reply docs/framework/unmanaged-api/debugging/icordebugmodule4-interface.md:54]: No, I'd say adjust these versions. These were surfaced on .NET 5

**Pull request merged**